### PR TITLE
Student Facing Lesson Plans: add option to script editor

### PR DESCRIPTION
**Pull request by JillianK**
This adds "Include student-facing lesson plans" to the script editor. It will not show up if the script is not migrated.
Screen Shot 2021-02-16 at 4 22 32 PM
Screen Shot 2021-02-16 at 4 22 26 PM

when isMigrated is false:
Screen Shot 2021-02-16 at 4 25 54 PM

## Links

- [spec](https://docs.google.com/document/d/1aydi_0ZG8rLZ15UeHBYN9g_uxwK0Uq9kcNwNf0oX2t4/edit)
- [jira](https://codedotorg.atlassian.net/browse/PLAT-313?atlOrigin=eyJpIjoiM2ZlM2I4M2UzMWE2NDA3YWI4MTIzYTBhZThhNmQ5MTgiLCJwIjoiaiJ9)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/levelbuilder/script-editor/ScriptEditor.jsx [hunk 1]
```diff
@@ -95,6 +95,7 @@ class ScriptEditor extends React.Component {
     initialShowCalendar: PropTypes.bool,
     initialWeeklyInstructionalMinutes: PropTypes.number,
     isMigrated: PropTypes.bool,
+    includeStudentFacingLessonPlans: PropTypes.bool,
```
Comment (dmcavoy) [inline apps/src/lib/levelbuilder/script-editor/ScriptEditor.jsx]: To help keep the difference between the state and the props clear we have been calling things that are also in the state initialIncludeStudentFacingLessonPlans

File: apps/src/lib/levelbuilder/script-editor/ScriptEditor.jsx [hunk 2]
```diff
@@ -153,7 +154,9 @@ class ScriptEditor extends React.Component {
       lessonDescriptions: this.props.i18nData.stageDescriptions,
       teacherResources: resources,
       hasImportedLessonDescriptions: false,
-      oldScriptText: this.props.initialLessonLevelData
+      oldScriptText: this.props.initialLessonLevelData,
+      includeStudentFacingLessonPlans:
```
Comment (dmcavoy) [inline apps/src/lib/levelbuilder/script-editor/ScriptEditor.jsx]: I think in order to make all the places we are talking about this shorter I think we can probably call them "student lesson plans" and leave out the facing part. This comment probably applies to all the files

File: apps/src/sites/studio/pages/scripts/edit.js [hunk 3]
```diff
@@ -78,6 +78,9 @@ export default function initPage(scriptEditorData) {
           scriptData.weeklyInstructionalMinutes
         }
         isMigrated={scriptData.is_migrated}
+        includeStudentFacingLessonPlans={
+          scriptData.includeStudentFacingLessonPlans
```
Comment (dmcavoy) [inline apps/src/sites/studio/pages/scripts/edit.js]: Can you move the default value setting out here instead of in the state for consistency?

**Review by dmcavoy — APPROVED** [review]
Totally thought I had already approved this. Ooops. Thanks for retagging me! This looks good to go

**Review by bethanyaconnor — APPROVED** [review]
🎉

**Pull request merged**